services/dialog/v1.js

```diff
@@ -0,0 +1,342 @@
+  // get profile variable - GETs the profile variables associated with the dialog
+
+  function WatsonDialogNode (config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+   
+    this.log("Watson Developer Cloud Contribution - Dialog Node Instantiated") ;
+	
+    this.on("input", function (msg) {
+      this.log("Watson Developer Cloud Contribution - Input received");
+		
+      var message = "";
+      if (!msg.payload) {
+			 message = "Missing property: msg.payload";
+			 node.error(message, msg);
+			 return;
+      }
+
+      var username = sUsername || this.credentials.username;
+      var password = sPassword || this.credentials.password;
+      this.status({});  
+			
+      if (!username || !password) {
+        this.status({fill:"red", shape:"ring", text:"missing credentials"});  
+        message = 'Missing Watson Dialog service credentials';
+        this.error(message, msg);
+        return;
+      } 
+
+      var dialog = watson.dialog({
+        username: username,
+        password: password,
+        version: "v1"
+      });	  
+
+      var params = {};
+      if (config.mode === "create") {
+        performCreate(node,dialog,msg);
+      }
+      else if (config.mode === "delete") {
+        performDelete(node,dialog,msg,config);
+      }
+      else if (config.mode === "deleteall") {
+        performDeleteAll(node,dialog,msg,config);
+      }
+       else if (config.mode === "list") {
+        performList(node,dialog,msg);
+      } else if (config.mode === "startconverse" || config.mode === "converse" || config.mode === "getprofile") {
+          var dialogid = config.dialog;
+          var clientid = config.clientid;
+          var converseid = config.converse;
+
+          if (!dialogid || "" == dialogid) {
+            if (msg.dialog_params && "dialog_id" in msg.dialog_params) {
+              dialogid = msg.dialog_params["dialog_id"];
+            }	
+          }
+		  
+          if (!dialogid || "" == dialogid) {
+            message = "Missing Dialog ID";
+            node.status({fill:"red", shape:"dot", text:message});	
+            node.error(message, msg);	
+          }			
+			
+          if (config.mode === "converse"  || config.mode === "getprofile") {
+            if (!clientid || "" == clientid) {
+              if (msg.dialog_params && "client_id" in msg.dialog_params) {
+                clientid = msg.dialog_params["client_id"];
+              }	
+            }				  
+            if (!converseid || "" === converseid) {
+              if (msg.dialog_params && "converse_id" in msg.dialog_params) {
+                converseid = msg.dialog_params["converse_id"];
+              }	
+            }				  
+		    if (!clientid || "" === clientid) {
+              message = "Missing Client ID";
+              node.status({fill:"red", shape:"dot", text:message});	
+              node.error(message, msg);	
+            }
+            if (!converseid || "" === converseid) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'message' is already declared in line 132.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2385269578/issues/source?bid=3225115&fileBranchId=3225116#l137)